src/capstone.rs

```diff
@@ -35,7 +51,22 @@ impl Instruction for capstone::Instr {
     }
 
     fn is_return(&self, _disassembler: &Disassembler) -> bool {
-        false
+        match self.detail {
+            None => false,
+            Some(ref detail) => {
+                match self.id {
+                    capstone::InstrIdArch::X86(_) => {
+                        match detail.groups.iter().find(|&&x| {
+                            x == x86_insn_group::X86_GRP_RET.as_int()
+                        }) {
+                            Some(_) => true,
+                            None => false,
+                        }
+                    }
+                    _ => false, // XXX Not supported yet
+                }
+            }
+        }
     }
```

Comment:
These functions looks quite similar now and do almost the same thing. Maybe one can do a generic function that takes in a filter instead to reduce duplication here?